numba/targets/arraymath.py

```diff
@@ -3666,6 +3666,48 @@ def _is_nonelike(ty):
     return (ty is None) or isinstance(ty, types.NoneType)
 
 
+@overload(np.array)
+def np_array(a, dtype=None, copy=True):
+
+    # developer note... keep this function (type_can_asarray) in sync with the
+    # accepted types implementations below!
+    if not type_can_asarray(a):
+        return None
+
+    impl = None
+    if isinstance(a, types.Array):
+        if _is_nonelike(dtype) or a.dtype == dtype.dtype:
+            def impl(a, dtype=None, copy=True):
+                if copy:
+                    return a.copy()
```

Comment:
Is this `np.ndarray.copy` or `np.copy` in practice? They have different defaults for 'order'. Consider the following:
```python
from numba import njit
import numpy as np

@njit
def foo(z):
    return np.array(z)

g = np.zeros((4, 5), order='F')
x = foo(g)
print(x)
print(x.flags)

x = foo.py_func(g)
print(x)
print(x.flags)
```
at present gives:
```
[[0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]]
  C_CONTIGUOUS : True
  F_CONTIGUOUS : False
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
[[0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]]
  C_CONTIGUOUS : False
  F_CONTIGUOUS : True  #<--------- NOTE NumPy preserves order.
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
```